runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -879,46 +873,32 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
             int systemDownTriggerCounter = 0;
 
             while (true) {
```

Comment:
The behavior changed from "probing" to a broadcast, this while(true) is unnecessary and can spin on exceptions that you want to fail right away. I think we should remove it, especially because getLatestLayout swallows all network exceptions.